projects/kit/components/input-number/input-number.component.ts

```diff
@@ -0,0 +1,255 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    computed,
+    inject,
+    Input,
+    signal,
+} from '@angular/core';
+import {toSignal} from '@angular/core/rxjs-interop';
+import {MaskitoDirective} from '@maskito/angular';
+import type {MaskitoOptions} from '@maskito/core';
+import {maskitoInitialCalibrationPlugin} from '@maskito/core';
+import {
+    maskitoCaretGuard,
+    maskitoNumberOptionsGenerator,
+    maskitoParseNumber,
+} from '@maskito/kit';
+import {TuiControl} from '@taiga-ui/cdk/classes';
+import {CHAR_HYPHEN, CHAR_MINUS} from '@taiga-ui/cdk/constants';
+import {TUI_IS_IOS, tuiFallbackValueProvider} from '@taiga-ui/cdk/tokens';
+import {tuiInjectElement} from '@taiga-ui/cdk/utils/dom';
+import {tuiIsSafeToRound} from '@taiga-ui/cdk/utils/math';
+import {TuiTextfieldDirective} from '@taiga-ui/core/components/textfield';
+import {TUI_DEFAULT_NUMBER_FORMAT, TUI_NUMBER_FORMAT} from '@taiga-ui/core/tokens';
+import {tuiFormatNumber} from '@taiga-ui/core/utils/format';
+import {tuiMaskito} from '@taiga-ui/kit/utils';
+
+import {TUI_INPUT_NUMBER_OPTIONS} from './input-number.options';
+
+const DEFAULT_MAX_LENGTH = 18;
+
+@Component({
+    standalone: true,
+    selector: 'input[tuiInputNumber]',
+    template: '',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [tuiFallbackValueProvider(null)],
+    hostDirectives: [
+        // TODO: replace with `TuiWithTextfield` after merging of https://github.com/taiga-family/taiga-ui/pull/9976
+        {
+            directive: TuiTextfieldDirective,
+            inputs: ['invalid', 'focused', 'readOnly', 'state'],
+        },
+        MaskitoDirective,
+    ],
+    host: {
+        '[value]': 'textfieldValue()',
+        '[disabled]': 'disabled()',
+        '[attr.inputMode]': 'inputMode()',
+        '[attr.maxLength]': 'maxLength()',
+        '(input)': 'onInput()',
+        '(blur)': 'onBlur()',
+        '(focus)': 'onFocus()',
+    },
+})
+export class TuiInputNumber extends TuiControl<number | null> {
+    private readonly element = tuiInjectElement<HTMLInputElement>();
+    private readonly isIOS = inject(TUI_IS_IOS);
+    private readonly options = inject(TUI_INPUT_NUMBER_OPTIONS);
+    private readonly numberFormat = toSignal(inject(TUI_NUMBER_FORMAT), {
+        initialValue: TUI_DEFAULT_NUMBER_FORMAT,
+    });
+
+    private readonly min = signal(this.options.min);
+    private readonly max = signal(this.options.max);
+    private readonly prefix = signal(this.options.prefix);
+    private readonly postfix = signal(this.options.postfix);
+
+    private readonly precision = computed(() =>
+        Number.isNaN(this.numberFormat().precision) ? 2 : this.numberFormat().precision,
+    );
```

Comment:
Should we make BC for this logic ? 🤔 

```suggestion
    private readonly precision = computed(() =>
        Number.isNaN(this.numberFormat().precision) ? 0 : this.numberFormat().precision,
    );
```